server/src/main/java/org/opensearch/identity/noop/RunAsSystemClient.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.identity.noop;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionType;
+import org.opensearch.client.Client;
+import org.opensearch.client.FilterClient;
+import org.opensearch.common.annotation.InternalApi;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.action.ActionResponse;
+
+/**
+ * Implementation of client that will run transport actions in a stashed context
+ * <p>
+ * This class and related classes in this package will not return nulls or fail permissions checks
+ *
+ * This class is used by the NoopIdentityPlugin to initialize IdentityAwarePlugins
+ *
+ * @opensearch.internal
+ */
+@InternalApi
+public class RunAsSystemClient extends FilterClient {
```

Comment:
This isn't an instance of a new client, its a wrapper around that local node client initialized in Node.java so overrides the `doExecute` method.

In particular, this is the default implementation that stashes the context prior to executing an actions and restores it prior to delegating back to the original actionListener's onResponse or onFailure.

Reply:
In the multitenant client mentioned earlier, the pattern is:
 - code stashes context
 - code calls the sdkClient (wrapper client)
 - for the NodeClient implementatoin, wrapper client then calls `client.foo()` which invokes the "protected" call

So for this to work we would need to change the default client implementation to use this `RunAsSystemClient` instance conditionally.   Not sure how we do that... I'm sure it's possible, though.

See https://github.com/opensearch-project/opensearch-remote-metadata-sdk/blob/main/core/src/main/java/org/opensearch/remote/metadata/client/impl/LocalClusterIndicesClient.java for existing implementation